lib/src/front_matter_document.dart

```diff
@@ -1,11 +1,85 @@
+import 'dart:io';
+
+import 'package:meta/meta.dart';
 import 'package:yaml/yaml.dart';
 
+import '../front_matter.dart';
+
+// Default delimiter for YAML.
+const defaultDelimiter = '---';
+
 /// Document containing the original `value`, front matter `data` and `content`.
 class FrontMatterDocument {
-  String value, content;
+  FrontMatterDocument({@required this.data, @required this.content});
+
+  FrontMatterDocument.fromText(String text,
+      {String delimiter = defaultDelimiter}) {
+    // Remove any leading whitespace.
+    final value = text.trimLeft();
+
+    // If there's no starting delimiter, there's no front matter.
+    if (!value.startsWith(delimiter)) {
+      content = value;
+      data = YamlMap();
+      return;
+    }
+
+    // Get the index of the closing delimiter.
+    final closeIndex = value.indexOf('\n' + delimiter);
+
+    // Get the raw front matter block between the opening and closing delimiters.
+    final frontMatter = value.substring(delimiter.length, closeIndex);
+
+    if (frontMatter.isNotEmpty) {
+      try {
+        // Parse the front matter as YAML.
+        data = loadYaml(frontMatter);
+      } catch (e) {
+        throw FrontMatterException(invalidYamlError);
+      }
+    }
+
+    // The content begins after the closing delimiter index.
+    content = value.substring(closeIndex + (delimiter.length + 1)).trim();
+
+    print('content: $content');
+  }
+
+  static fromFile(String path, {String delimiter = defaultDelimiter}) async {
+    var file = File(path);
+
+    // Throw an error if file not found.
+    if (!await file.exists()) {
+      throw FrontMatterException(fileNotFoundError);
+    }
+
+    try {
+      var text = await file.readAsString();
+      return FrontMatterDocument.fromText(text, delimiter: delimiter);
+    } catch (e) {
+      // Handle downstream errors, or throw one if file is not readable as text.
+      switch (e.message) {
+        case invalidYamlError:
+          rethrow;
+        default:
+          throw FrontMatterException(fileTypeError);
+      }
+    }
+  }
+
+  String content;
 
   /// The parsed YAML front matter as a [YamlMap].
   YamlMap data;
 
-  FrontMatterDocument(this.value);
+  String get value {
```

Comment:
The reason for the getter is so that the document can be modified and converted back to a raw String to be stored. Eg. I'm planning to use it with Git storage.

Reply:
I've just released `1.1.0` with a `toString()` method on `FrontMatterDocument` that returns the raw, original value. Is this what you need?

Perhaps we should pause this PR and open the discussion in the issue #1 ?

Reply:
No. Maybe I'm communicating it wrong, but the point of the getter is so that one can modify the `data` or `content` variables on a `FrontMatterDocument` can be modified and easily converted back to a raw front-matter document.